### DISCUSSION TITLE
Installation sous Colab

### DISCUSSION THREAD

**Discussion by DamienKoala** [IGNF/myria3d#73] [Q&A] [answered]
Bonjour,
Je tente d'installer Myria3D sous Colab avec Conda. J'ai copié le "environment.yml" tel que proposé sous github. Par contre, lors de l'activation de Myria3D j'ai un message d'erreur
` "/usr/local/bin/activate: 6: /usr/local/envs/myria3d/etc/conda/activate.d/gdal-activate.sh: [[: not found".`
Je ne suis pas doué sur les environnements virtuels, qui plus est sous Colab...
J'ai tenté de pousuivre malgré tout avec pip, et re-message d'erreur
```
× python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
    note: This error originates from a subprocess, and is likely not a problem with pip.
  Preparing metadata (setup.py) ... error
error: metadata-generation-failed
```
Avez-vous une piste pour me permettre de poursuivre ?
Merci
Damien

**Comment by CharlesGaydon** [comment]
Bonjour,
Cela semble être un problème lié à l'installation de GDAL, [qui est une dépendance de PDAL](https://pdal.io/en/latest/development/compilation/dependencies.html#gdal-3-0).
Dans un premier temps je vous suggère d'isoler le problème pourrait être de créer un environnemnet virtuel minimal avec [python-pdal](https://github.com/IGNF/myria3d/blob/a739e007aaa3408662086c3a9e98e8d36928a78b/environment.yml#LL19C12-L19C12) seulement. Si le problème persiste bien, vérifier ensuite que la situation n'émerge que dans Google Collab, en tentant l'installation de l'environnement conda en local (si possible).
Sur le problème lui même, je ne peux que suggérer des pistes : soit (1) gdal n'est pas trouvé, soit (2) gdal n'est pas installé. 

Pour tester (2) la commande `gdal-config --version` peut-être une option. Si gdal n'est pas installé, une installation manuelle avec `sudo apt-get install libgdal-dev` débloquera peut-être la situation. (source [gis-stackexchange](https://gis.stackexchange.com/a/124420/184486)).

N'hésitez pas à me tenir au courant, et si l'installation sous Collab nécessite une manoeuvre spécifique nous pourrons la documenter avec un google Collab public partagé ! :)

**Reply by CharlesGaydon** [reply]
Bonjour @DamienKoala, avez-vous eu plus de succès sous Collab ?

**Reply by DamienKoala** [reply]
Bonjour,
Je n'ai pas trop eu le temps de me repencher sur le sujet... Cela étant, dans environnement.yml j'ai ajouté GDAL dans les dépendances, ce qui permet de vérifier la version installée via !gdal-config --version (3.3.2).
Dans les log d'installation, il y a ce message d'alerte :
```
WARNING:
    You currently have a PYTHONPATH environment variable set. This may cause
    unexpected behavior when running the Python interpreter in Mambaforge.
    For best results, please verify that your PYTHONPATH only points to
    directories of packages that are compatible with the Python interpreter
    in Mambaforge: /usr/local
```
Vive les environnements virtuels :) ! Le PYTHONPATH actuel est '/env/python'
Faut-il donc pointer le PYTHONPATH vers '/usr/local' ?

Autre chose, le message d'erreur, "[[: not found" semble renvoyer vers un soucis de bash/sh lors de l'execution de gdal-activate.sh...
Cette [manip](https://stackoverflow.com/questions/62610289/conda-activate-and-conda-init-fail-to-work-in-colab-as-per-june-2020) semble régler le sujet (pas d'erreur sur conda activate myria3d) ? Il faut désormais tester Myria3d pour voir si le sujet est clos 

```
%%shell
eval "$(conda shell.bash hook)" # copy conda command to shell
conda activate myria3d
python --version
conda deactivate
```

Merci,
Damien

**Reply by CharlesGaydon** [reply]
Merci pour ce retour.
>  Faut-il donc pointer le PYTHONPATH vers '/usr/local' ?
Peut-être, ça semble être l'approche suggérée...
N'hésite pas à me tenir au courant dans cette discussion, je la laisse ouverte pour l'instant.
Si ton Notebook est partageable n'hésite pas à en fournir un lien ici, et je pourrai peut-être (sans garantie) y jeter un coup d'oeil jeudi :)

**Reply by DamienKoala** [reply]
Bonjour,
Voici le lien vers Colab : [https://colab.research.google.com/drive/1Y5cANVJntG-pmvfAeed6eZ0Sl1796jHi?usp=sharing](url)
J'ai tenté de suivre les consignes d'installation. Il y a quelques commandes parasites pour connaître les version, répertoire etc.
A suivre...
Merci à toi
_PS, j'ai suivi le webinaire mardi de l'IGN sur le sujet, que j'ai trouvé inintéressant ! Les présentations sont très orientées "QGIS" car en effet beaucoup d'utilisateurs ont recours à cette solution. Je reste sur une approche Python, peut-être plus ardue, car je ne connais pas Qgis, et aussi parce que l'exploitation des données me semble (à tord ?) plus simple... (extraire des surfaces de végétation basse, haute etc)._
Damien

**Reply by CharlesGaydon** [reply, marked as answer]
Bonjour @DamienKoala,
En jetant un coup d'oeil à ton Collab, je vois que le code de myria3d semble tout de même pouvoir utiliser directement depuis ses sources, en renplacant la dernière cellule par
```
  !git clone https://github.com/IGNF/myria3d
  !source activate myria3d && python /content/myria3d/run.py -h
```
